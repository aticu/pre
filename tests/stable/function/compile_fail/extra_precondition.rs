use pre::{check_pre, pre};

#[pre(condition("must be bar"))]
fn foo() {}

#[check_pre]
fn main() {
    #[assert_pre(condition("must be bar", reason = "is bar"))]
    #[assert_pre(condition("must be baz", reason = "is baz"))]
    foo()
}
