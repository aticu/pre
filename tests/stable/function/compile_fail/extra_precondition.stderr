error[E0560]: struct `foo` has no field named `_custom_must_20be_20baz`
 --> $DIR/extra_precondition.rs:9:14
  |
9 |     #[assure("must be baz", reason = "is baz")]
  |              ^^^^^^^^^^^^^ `foo` does not have this field
  |
  = note: available fields are: `_custom_must_20be_20bar`
